src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs

```diff
@@ -96,80 +96,102 @@ protected Task<ImmutableArray<CodeActionOperation>> GetPreviewOperationsAsync(Ca
 
         public void Invoke(CancellationToken cancellationToken)
         {
-            SourceProvider.UIThreadOperationExecutor.Execute(CodeAction.Title, CodeAction.Message, allowCancellation: true, showProgress: true, action: context =>
+            // Fire and forget.  The called method will set up async operation tracking synchronously on 
+            // this thread.
```

Comment:
What happens if this async stuff underneath doesn't actually yield the UI thread -- are we setting up a threaded wait dialog atop the editor's one?